### PR TITLE
created the sam deployer

### PR DESCRIPTION
**Pull request by MatteoIorio11**

### CODE REVIEW

File: .github/workflows/aws_sam.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Deploy the lambda using AWS SAM
+"on":
+  push:
+    branches:
+      - aws-sam
+  workflow_dispatch: {}
+jobs:
+  deploy:
+    name: deploy
+    runs-on: ubuntu-latest
```
Comment (pavlovic-ivan) [inline .github/workflows/aws_sam.yml:10]: I would suggest adding an environment here, called `protected`. This adds additional level of security in the workflow

File: .github/workflows/aws_sam.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - run: npm ci
+      - uses: aws-actions/setup-sam@v2
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - run: sam build
+      - run: >
+          sam deploy --stack-name ${{ secrets.AWS_STACK_NAME }} \
```
Comment (pavlovic-ivan) [inline .github/workflows/aws_sam.yml:27]: Move this to the Makefile, and refactor this step to simply run a `make` command (check how it is done [here](https://github.com/pavlovic-ivan/ghrunner-app-gcp/blob/main/.github/workflows/deploy.yml#L27))

**Comment by pavlovic-ivan** [comment]
@MatteoIorio11 looks good in general. However, please address the comments made in the PR, and also add missing trigger logic (see [here](https://github.com/pavlovic-ivan/ghrunner-app-gcp/blob/main/app.js#LL11C9-L11C9)).

File: .github/workflows/aws_sam.yml (hunk 2 above)
Comment (pavlovic-ivan) [reply .github/workflows/aws_sam.yml:27]: Also, don't forget that the stack name and function name should be defined similar as in the Makefile (`gcloud beta functions deploy ${name}`)

Comment (MatteoIorio11) [reply .github/workflows/aws_sam.yml:27]: Hi Ivan, I am working on the Makefile and I have question. Should I have to set the  "--s3-bucket" as the "--stack-name" and the function name (create a var inside the makefile and set the --s3-bucket as :  --s3-bucket ${bucket} or use an env variable as the others tags)??

**Comment by MatteoIorio11** [comment]
I have added the 'sam build' and the 'sam deploy' inside the make and I have set the "stack-name", "the function name" and the "s3-bucket" defined in the top of the makefile

**Comment by MatteoIorio11** [comment]
I have also removed from the "setup.md" the information about the function name, stack name and the s3 bucket because are hard coded inside the makefile

**Review by pavlovic-ivan — APPROVED** [review]
Thanks @MatteoIorio11 . great job 🚀

**Comment by MatteoIorio11** [comment]
Thank you @pavlovic-ivan for the feedback, for every bug or problem feel free to contact me

**Pull request merged**